docs/data/data-grid/filtering/CustomMultiValueOperator.tsx

```diff
@@ -124,15 +123,21 @@ export default function CustomMultiValueOperator() {
     ],
   });
 
-  if (columns.length > 0) {
-    const quantityColumn = columns.find((col) => col.field === 'quantity');
-    const newQuantityColumn = {
-      ...quantityColumn!,
-      filterOperators: quantityOnlyOperators,
-    };
-    const quantityColIndex = columns.findIndex((col) => col.field === 'unitPrice');
-    columns[quantityColIndex] = newQuantityColumn;
-  }
+  const columns = React.useMemo(() => {
+    const newColumns = [...data.columns];
+
+    if (newColumns.length > 0) {
+      const index = newColumns.findIndex((col) => col.field === 'quantity');
+      const quantityColumn = newColumns[index];
+
+      newColumns[index] = {
+        ...quantityColumn,
+        filterOperators: quantityOnlyOperators,
```

Comment:
This demo had two `quantity` columns. It was keeping the same `field` for both. There's a "jump" when scrolling horizontally if two columns have the same field.